src/components/QrcodeStream.vue

```diff
@@ -133,22 +133,63 @@ export default {
   },
 
   methods: {
+    async setCameraById(deviceId) {
+      this.cameraInstance?.stop();
+      this.cameraInstance = await Camera(this.$refs.video, {
+        camera: null,
+        deviceId,
+        torch: this.torch
+      });
+
+      localStorage.setItem("scanner-device-id", deviceId);
+    },
+    async listAvailableCameras() {
+      return await getAvailableCameras();
+    },
+    async reloadCamera() {
+      if (this.cameraInstance) {
+        this.cameraInstance.stop();
+      }
+
+
+      this.cameraInstance = await Camera(this.$refs.video, {
+        camera: this.camera,
+        torch: this.torch,
+      });
+
+
+      if (this.shouldScan) {
+        this.startScanning();
+      }
+
+      console.log("Camera reloaded successfully.");
+    },
     init() {
       const promise = (async () => {
         this.beforeResetCamera();
 
+
         if (this.camera === "off") {
           this.cameraInstance = null;
 
           return {
             capabilities: {}
           };
         } else {
-          this.cameraInstance = await Camera(this.$refs.video, {
-            camera: this.camera,
-            torch: this.torch
-          });
-
+          const preferredDeviceId = localStorage.getItem('scanner-device-id');
```

Comment:
I think we shouldn't mess with local storage inside the library. What if the user is already using the key `scanner-device-id` for something else? But more importantly, I think this has legal implications at least in the EU. If you store any data, you need to show a cookie banner. I think the library should force that upon users.